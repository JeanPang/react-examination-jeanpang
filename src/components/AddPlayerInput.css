.form_control {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
  /* width: 300px; */
  width: 75%;
}

.formAddPlayerInput {
  display: flex;
  justify-content: space-between;
  margin: 5px;
}

.formAddPlayerInput > input {
  display: inline;
  border-radius: 2px;
  border: none;
  padding-left: 5px;
  padding-right: 5px;
}

input:focus,select:focus {
  border: none;
  outline: none;
  box-shadow: none;
}

.btn_select {
  width: 50px;
  text-indent: 5px;
  background: white;
  border: none;
}

.btn_add {
  padding-left: 15px;
  padding-right: 15px;
}

.btn_add:active {
  background: #abaaaa;
}