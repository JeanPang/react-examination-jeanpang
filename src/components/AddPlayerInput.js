import React, { Component } from 'react';
import './AddPlayerInput.css';

class AddPlayerInput extends Component {
  render() {
    return (
      <form 
        onSubmit={this.submitForm} 
        className="formAddPlayerInput"
      >
        <input
          type="text"
          name="name"
          autoFocus={true}
          className="form_control"
          placeholder=" Type the name of a player "
          value={this.state.formControls.name} 
          onChange={(e) => {
            this.handleChange(e);
          }}
        />

        <select 
          name="position"  
          value={this.state.formControls.position} 
          onChange={(e) => {
            this.handleChange(e);
          }}
          className="btn_select"
        >
        　<option value="SF">SF</option>
        　<option value="PG">PG</option>
        </select>

        <input 
          type="submit" 
          value="add"
          className="btn_add"
        />
      </form>
    );
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      formControls: {
          name: this.props.name || '',
          position: 'SF',
      }
  }
  }

  submitForm=(e)=>{
    e.preventDefault()
    if (this.state.formControls.name.length !== 0){
      this.props.addPlayer(this.state.formControls.name, this.state.formControls.position);
    }

    this.setState({ 
      formControls: {
        name: '',
        position: 'SF',
      }
    });
  }

  handleChange=(e)=> {
    const name = e.target.name;
    const value = e.target.value;
    this.setState({
      formControls: {
          ...this.state.formControls,
          [name]: value,
      }
    });
  }
}

export default AddPlayerInput;
