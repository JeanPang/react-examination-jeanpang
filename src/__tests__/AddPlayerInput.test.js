import React from 'react';
import AddPlayerInput from '../components/AddPlayerInput';
import renderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';

const player = [
  {
    name: 'LeBron James',
    team: 'LOS ANGELES LAKERS',
    position: 'SF',
    starred: true,
  },
  {
    name: 'Kevin Duran',
    team: 'GOLDEN STATE WARRIORS',
    position: 'SF',
    starred: false,
  },
  {
    name: 'Anthony Davis',
    team: 'NEW ORLEANS PELICANS',
    position: 'PF',
    starred: false,
  },
  {
    name: 'Stephen Curry',
    team: 'GOLDEN STATE WARRIORS',
    position: 'PG',
    starred: false,
  },
  {
    name: 'James Harden',
    team: 'HOUSTON ROCKETS',
    position: 'SG',
    starred: false,
  },
  {
    name: 'Kawhi Leonard',
    team: 'TORONTO RAPTORS',
    position: 'SF',
    starred: false,
  },
]

describe('AddPlayerInput', function() { 
  var AppElement = TestUtils.renderIntoDocument(<AddPlayerInput addPlayer={player}></AddPlayerInput>); 
  var DomElement = ReactDOM.findDOMNode(AppElement); 
  var input = DomElement.getElementsByTagName('input')[0]; 
  var submitBtn = DomElement.getElementsByTagName('input')[1]; 

  it('renders correctly', () => {
    const tree = renderer
      .create(<AddPlayerInput addPlayer={player}></AddPlayerInput>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('type should be text', function() { 
   expect(input.getAttribute('type')).toEqual('text'); 
  }); 

  it('should render 2 input', () => {
    const testRenderer = renderer.create(<AddPlayerInput />);
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node => node.type === 'input')).toHaveLength(2);
  });

  it('should empty name and set default position to SF', function() {
    TestUtils.Simulate.click(submitBtn);
    expect(AppElement.state.formControls).toEqual({
        name: "",
        position: "SF",
    });
  });
}); 
