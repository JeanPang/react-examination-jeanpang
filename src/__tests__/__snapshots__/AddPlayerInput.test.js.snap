// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlayerInput renders correctly 1`] = `
<form
  className="formAddPlayerInput"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="form_control"
    name="name"
    onChange={[Function]}
    placeholder=" Type the name of a player "
    type="text"
    value=""
  />
  <select
    className="btn_select"
    name="position"
    onChange={[Function]}
    value="SF"
  >
    　
    <option
      value="SF"
    >
      SF
    </option>
    　
    <option
      value="PG"
    >
      PG
    </option>
  </select>
  <input
    className="btn_add"
    type="submit"
    value="add"
  />
</form>
`;
